languages/rust/exercises/concept/string-vs-str/.meta/design.md

```diff
@@ -0,0 +1,50 @@
+# Goal
+
+Teach the student the differences between String and &str and how to work successfully with both.
+Learning objectives
+
+    Know String is owned and mutable
+    Know &str is shared and immutable
+    Know String has &str methods because of shared traits
+    Know of performance characteristics
+
+## Out of scope
+
+    Machine level representation of either
+    in depth performance differences
+
+## Concepts
+
+    mutability
+    ownership
+    slices
+
+## Prerequisites
+
+    basics
+    slices
+    str-use
+    string-use
+
+## Resources to refer to
+
+## Hints
+
+    [string slices discussion in TRPL book][https://doc.rust-lang.org/book/ch04-03-slices.html#string-slices]
```

Comment:
nitpick: could we make these links consistently formatted? 
```md
[string slices discussion in TRPL book](https://doc.rust-lang.org/book/ch04-03-slices.html#string-slices)
[&str primitive documentation](https://doc.rust-lang.org/std/primitive.str.html)
[std::str](https://doc.rust-lang.org/std/str/)
[string types in RBE](https://doc.rust-lang.org/stable/rust-by-example/std/str.html)
```
should do it.
Thanks @coriolinus for noting my earlier incorrect suggestion. Didn't mean to delete your comment FWIW; GitHub seems to do that for deleting the original suggestion.